### DISCUSSION TITLE
What is the goal of this benchmark?

### DISCUSSION THREAD

**Discussion by saattrupdan** [EuroEval/EuroEval#85] [General]
This is a general discussion about the aims of this benchmark in general. Are we aiming to keep all the datasets across languages consistent with each other, to facilitate comparison between languages? Alternatively, are we aiming to make each dataset as challenging as possible, by making the domains covered in each as diverse as possible (for instance, by making the summarisation task cover both news articles, research articles, and more)? 

The original aim of the benchmark was to measure _general pretrained language model performance_, and as such not measure "the best NER model", for instance. This led to the decision of minimising the training dataset size, to emphasise pretraining knowledge. Has this assumption changed? And regarding the benchmarking of generative models, is this still the goal?

@KennethEnevoldsen @peter-sk @peterbjorgensen

**Comment by KennethEnevoldsen** [comment]
> The original aim of the benchmark was to measure general pretrained language model performance, and as such not measure "the best NER model", for instance. This led to the decision of minimising the training dataset size, to emphasise pretraining knowledge. Has this assumption changed? 

No def. agree with this assumption

> And regarding the benchmarking of generative models, is this still the goal?

I would say yes.

> This is a general discussion about the aims of this benchmark in general. Are we aiming to keep all the datasets across languages consistent with each other, to facilitate comparison between languages? Alternatively, are we aiming to make each dataset as challenging as possible, by making the domains covered in each as diverse as possible (for instance, by making the summarisation task cover both news articles, research articles, and more)?

As it possible now to make comparisons between languages? I feel like this is already quite hard e.g. due to the consistency of annotation or similar. 

One problem I could see is that if we e.g. make all the Danish tasks more challenging it might seem like the models are just worse at Danish.

**Reply by saattrupdan** [reply]
> One problem I could see is that if we e.g. make all the Danish tasks more challenging it might seem like the models are just worse at Danish.

I agree that aggregation would need to change. Currently we're simply averaging the Danish/Norwegian/Swedish scores, but this would not be possible if the scores are substantially different. A potential remedy here is to compute the average language _rank_ rather than score?

**Reply by KennethEnevoldsen** [reply]
yea, Lasse and I also discussed rank as a better metric than means. It make it relative to other models though. Another option would be to scale it pr. language such that each language score is between 0-1?

**Reply by KennethEnevoldsen** [reply]
The flipside is that ScandEval atm. might be biased toward news and social media and thus when training a new model we might be biased toward using more news and social media at the cost of other domains.

**Reply by saattrupdan** [reply]
Yeah agreed, using ranks would mean that a model's score depends on other models in the benchmark and is therefore not static. But wouldn't scaling per language cause the same issue? Since what does "1" mean here, aside from the current best score? Unless we do a human evaluation and set that to "1", I guess.

**Reply by saattrupdan** [reply]
Regarding domains, maybe this is where adding language modelling would help? If we include a bunch of different datasets (maybe simply your DadsBench collection of datasets for Danish, if they're sharable) and compute ppl on each of them? We'd have to decide on how much weight is put on that in that case though.

**Reply by KennethEnevoldsen** [reply]
> But wouldn't scaling per language cause the same issue?

def. I actually don't think it is a very good idea (was just thinking out loud). One option is to do both mean and rank then you can just sort by the one you prefer.

> Regarding domains, maybe this is where adding language modelling would help? If we include a bunch of different datasets (maybe simply your DadsBench collection of datasets for Danish, if they're sharable) and compute ppl on each of them? We'd have to decide on how much weight is put on that in that case though.

but if we do that, wouldn't we want to do so for the other datasets as well?

**Reply by saattrupdan** [reply]
I think choosing between either mean or rank is not preferable, otherwise there would always be a discussion of which model is actually better. And I'm leaning towards mean in that case, since that allows us to transfer confidence intervals, meaning that we can have "joint first place" models, for instance.

**Reply by KennethEnevoldsen** [reply]
I prefer to mean as well (ideally you are really interested in a latent factor, sort of an LMM IQ). However, the mean is so much simpler so I think we should keep it.

**Reply by peterbjorgensen** [reply]
I agree that scores across languages might be difficult and I find ranks hard to interpret because it doesn't make sense when models are close in performance. Another solution could be to have normalised scores compared to some base model, so you can say that "this model is 10 percent better at Norwegian than this model". It will be difficult to compare a single model across languages. I think we should strive towards getting the best possible dataset for each language rather than 1-to-1 correspondence between different languages. E.g. it might be that some country has high quality high school assignments available online - then it's better to use those than translate to each language from English.

**Reply by KennethEnevoldsen** [reply]
Relative improvement is quite interesting as it accounts for difficulty of the task prior to averaging. XLM-roberta-base could be a good candidate, but it would give some odd results for the smaller models as they would have a negative score. That might be acceptable.

**Reply by saattrupdan** [reply]
Relative improvement assumes that the base model is equally good at all the languages though, so we'd have to be careful with the choice of model. But otherwise sounds like a good idea 🙂

**Reply by KennethEnevoldsen** [reply]
Alternatively, an option is to select the best possible dataset and then report a mean (same as usual but removing the correspondence criteria).  Just to outline what bias that might introduce:

Lets say a balanced model would gain the scores:
- 0.9 (Sw), 0.9 (No), 0.6 (Da)
Indicating that Danish was the hardest language (either through harder tasks, inconsistent tagging, or just a more difficult language for LLMs).

The we imagine 2 new models:
- A better multilingual model (moderate improvement across all)
  - 0.95 (Sw), 0.95 (No), 0.65 (Da)
- A better danish model (only improve Danish)
  -  0.90 (Sw), 0.90 (No), 0.75 (Da)

Both of these would lead to a similar increase in performance, which would lead developers (who wants to compete on the benchmark) to develop more on the "harder" language. This in turn causes and incentive to increase the difficulty of ones own tasks. Note: This changes nothing pr. language and just having "inconsistent" datasets (impossible to gain a better score on) would not lead to the same priority.

**Comment by saattrupdan** [comment]
Regarding adding new leaderboards, I'm currently thinking of having the following:

1. Mainland Scandinavian NLU
2. Mainland Scandinavian NLG 
3. Insular Scandinavian NLU
4. Insular Scandinavian NLG
5. German NLU
6. German NLG
7. Dutch NLU
8. Dutch NLG
9. English NLU
10. English NLG

The NLU benchmarks are like the current one, and the NLG benchmarks only contain benchmarks of generative models, and is done on all the NLU and NLG tasks, where the NLU tasks are framed as generative tasks. Thus, there would be a substantial overlap between the NLU and NLG benchmarks.

This is not set in stone at all; it's merely a suggestion. The idea is that since the analysis in the ScandEval paper showed that it doesn't make sense to compare Mainland Scandinavian performance with German/Dutch/English/Icelandic/Faroese, it would be misleading to include them in the same leaderboard. The same analysis showed that it makes sense to merge Icelandic and Faroese. And maybe it would also show that German and Dutch should be merged. Don't know about English.

I guess one benefit of having separate leaderboards in this way is that it gives a bit more freedom, as we're not supposed to compare across different leaderboards. So, for instance, if German had a special task not present in the other languages, they could include that in their benchmark.

**Reply by saattrupdan** [reply]
Alternatively, it could be just two leaderboards: an NLU benchmark and an NLG benchmark, but where there are tabs for the language groups, like in [SEB](https://kennethenevoldsen.github.io/scandinavian-embedding-benchmark/). Notably, of course, there would be no "All" category.

**Reply by KennethEnevoldsen** [reply]
language tabs gives a nice overview. Potentially it might be relevant to create an "overview" tab with meaningful aggregations:

ALL (germanic?), mainland S, insular S, North germanic, west germanic
